components/operator/internal/controllers/stack/delete/brokers.go

```diff
@@ -0,0 +1,68 @@
+package delete
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/formancehq/operator/apis/stack/v1beta3"
+	"github.com/formancehq/operator/internal/controllers/stack/storage/nats"
+	"github.com/go-logr/logr"
+)
+
+var (
+	natsClientId = "membership"
+)
+var (
+	ErrCast = errors.New("cannot cast interface to string")
+)
+
+func DeleteByBrokers(c *v1beta3.Configuration, stackName string, subjectService []string, logger logr.Logger) error {
+	values := reflect.ValueOf(c.Spec.Broker)
+	for i := 0; i < values.NumField(); i++ {
+		if rv := reflect.ValueOf(values.Field(i)); !rv.IsValid() || rv.IsNil() {
+			continue
+		}
+		switch t := values.Field(i).Interface().(type) {
+		case v1beta3.NatsConfig:
```

Comment:
We handle only NATS ? Should we not handle kafka as well if clients want to use kafka ? (I don't know, just asking the question)

Reply:
Hum, that's a question, at begining i was asked to only do NATS, I can also do it for Kafka !